strawberryfields/backends/states.py

```diff
@@ -814,6 +815,43 @@ def expand_dims(op, n, modes):
 
         return mean, var
 
+    def number_expectation(self, modes):
```

Comment:
Might be worth renaming `modes` such it somehow describes the difference between modes of the state and modes for the expectation value (maybe `active_modes` or so). I got a bit confused with grasping the difference between `range(len(modes))` and `num_modes` later in the method.

Reply:
by parsimony I prefer Josh's answer.